File: frontend/src/microservices/attraction/objects/AttractionResponse.ts
// 本文件定义 attraction 模块的 `AttractionResponse`，作为响应数据并提供 JSON 编解码。

import type { AttractionTicketTypeResponse } from './AttractionTicketTypeResponse'

export type AttractionResponse = {
  attractionId: string
  attractionName: string
  city: string
  location: string
  description: string
  imageUrl: string | null
  status: string
  ticketTypes: AttractionTicketTypeResponse[]
}
export const attractionResponseFromJson = (json: string): AttractionResponse =>
  JSON.parse(json) as AttractionResponse

export const attractionResponseToJson = (value: AttractionResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/attraction/objects/AttractionResponse.scala
package com.typesafe.travel.attraction.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class AttractionResponse(
    attractionId: String,
    attractionName: String,
    city: String,
    location: String,
    description: String,
    imageUrl: Option[String],
    status: String,
    ticketTypes: List[AttractionTicketTypeResponse]
)
object AttractionResponse:
  given sourceEncoder: Encoder[AttractionResponse] = deriveEncoder
  given sourceDecoder: Decoder[AttractionResponse] = deriveDecoder
